src/shared/cpu-set-util.h

```diff
@@ -48,3 +48,36 @@ int cpu_set_to_dbus(const CPUSet *set, uint8_t **ret, size_t *allocated);
 int cpu_set_from_dbus(const uint8_t *bits, size_t size, CPUSet *set);
 
 int cpus_in_affinity_mask(void);
+
+typedef enum NUMAPolicyType {
+        NUMA_POLICY_DEFAULT,
+        NUMA_POLICY_PREFER,
+        NUMA_POLICY_BIND,
+        NUMA_POLICY_INTERLEAVE,
+        NUMA_POLICY_LOCAL,
+        _NUMA_POLICY_MAX,
+        _NUMA_POLICY_INVALID = -1,
+} NUMAPolicyType;
```

Comment:
this type appears redundant, no? sure, it adds a type name to the set of MPOL_xyz values defined in missing_syscall.h, but other than that it doesn't do much, no? i.e. we don't add anything useful here (besides the typedef name), hence I think we should probably not have this, but just use the kernels own type here (i.e. `int`), even if it's not as typesafe.

Reply:
Well, we use the type in bunch of places. We have translation table for to/from string conversions defined on it, accompanying helper function prototypes and such.

I think it is cleaner from the design perspective to have two types separated and have just a single place where we translate from our representation to low level kernel type, even when map operation is trivial.

Reply:
but why have both sets of enums? it's not worth it. Yes, it would have been great if the kernel's own definition would have used a proper enum type for this, but they don't. Let's accept that, and simple use the type they picked. In no other case we have second guessed the kernel's type and introduced our own type that the same but also not. I mean, it's stupid that on UNIX fds are "int"s, instead of their own type fd_t. But it's how it is, we have to follow it, we don't define our own fd_t, but just use "int", because that's what the kernel returns.

Note that the string translation stuff works the same for enums and for true ints. We use that various places, look at sched_policy_to_string() and friends for example. In fact, the sched policy stuff is almost the same as the numa policy case, it's also an int, with only a handful of values defined. Please copy the same concepts, including adding mpol_is_valid() or so, copying what sched_policy_is_valid() currently provides...